pages/add/add.jsx

```diff
@@ -31,11 +38,117 @@ export default React.createClass({
     event.preventDefault();
     this.setState({numCreatorFields: this.state.numCreatorFields+1});
   },
+  handleFormUpdate(evt, name, field, value) {
+    let formValues = this.state.formValues;
+
+    if ( name === `tags` ) {
+      value = value.split(`#`).map((tag) => {
+        return tag.trim();
+      });
+    }
+
+    formValues[name] = value;
+    this.setState({ formValues });
+  },
+  handleFormSubmit(event) {
+    event.preventDefault();
+    this.refs.basicForm.validates(basicFormIsValid => {
+      this.refs.detailForm.validates(detailFormIsValid => {
+        if (!basicFormIsValid) {
+          console.error(`basic form does not pass validation!`);
+        }
+
+        if (!detailFormIsValid) {
+          console.error(`detail form does not pass validation!`);
+        }
+
+        if (!basicFormIsValid || !detailFormIsValid) {
+          return;
+        }
+
+        this.postEntry(this.state.formValues);
+      });
+    });
+  },
+  getNonce(callback) {
+    Service.nonce()
+      .then((nonce) => {
+        callback(nonce);
+      })
+      .catch((reason) => {
+        this.setState({
+          authError: true
+        });
+        console.error(reason);
+        callback(false);
```

Comment:
why does this need to happen? if the callback needs to work for success and failure, it might make more sense to ensure the callback is a dual `callback(error, data)` function:
```
.then(nonce => {
  callback(false, nonce);
})
.catch(reason => {
  this.setState({ authError: true });
  callback(new Error("could not retrieve data from /nonce"));
});
```